### PR TITLE
Update paging methods to be opt in for returning count information

### PR DESCRIPTION
**Pull request by robertbennett1998**
Currently any paging requests (in this package) also make a count request, this can be very expensive to run and therefore shouldn't be ran be default but should be opt in.

https://stackoverflow.com/questions/27745346/get-record-count-in-azure-documentdb
> For example from the above thread: 
16,400 records in single partition consumes 535 RUs.  
[Youngjae](https://stackoverflow.com/users/361100/youngjae)
 [Jul 24, 2017 at 4:57](https://stackoverflow.com/questions/27745346/get-record-count-in-azure-documentdb#comment77508424_43560975)

# Breaking Changes
- Added default variable to opt out (by default) or returning the count information from paging queries.
    - This will require users who wish to get total count information to add an additional parameter to their calls.

### CODE REVIEW

File: src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs [hunk 1]
```diff
@@ -379,26 +380,29 @@ public async ValueTask<int> CountAsync(CancellationToken cancellationToken = def
...
                                                            _repositoryExpressionProvider.Default<TItem>()));
 
             _logger.LogQueryConstructed(query);
 
-            Response<int> countResponse = await query.CountAsync(cancellationToken);
+            Response<int>? countResponse = null;
+            if (returnTotal)
+            {
+                countResponse = await query.CountAsync(cancellationToken);
+            }
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs:394]:
```suggestion
            Response<int>? countResponse = null;
            
            if (returnTotal)
            {
                countResponse = await query.CountAsync(cancellationToken);
            }
```

File: src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs [hunk 2]
```diff
@@ -379,26 +380,29 @@ public async ValueTask<int> CountAsync(CancellationToken cancellationToken = def
...
 
             _logger.LogQueryExecuted(query, result.Charge);
 
             return new Page<TItem>(
-                countResponse.Resource,
+                countResponse?.Resource ?? null,
                 pageSize,
                 result.Items.AsReadOnly(),
-                result.Charge + countResponse.RequestCharge,
+                result.Charge + countResponse?.RequestCharge ?? 0,
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs:405]: Does this still need to sum up the charge from the count and the query?

File: src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs [hunk 3]
```diff
@@ -416,8 +421,11 @@ public async ValueTask<int> CountAsync(CancellationToken cancellationToken = def
                 .Where(_repositoryExpressionProvider
                     .Build(predicate ?? _repositoryExpressionProvider.Default<TItem>()));
 
-            Response<int> countResponse =
-                await query.CountAsync(cancellationToken).ConfigureAwait(false);
+            Response<int>? countResponse = null;
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs:424]:
```suggestion
            Response<int>? countResponse = null;
            
```

File: src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs [hunk 4]
```diff
@@ -430,11 +438,11 @@ public async ValueTask<int> CountAsync(CancellationToken cancellationToken = def
             _logger.LogQueryExecuted(query, result.Charge);
 
             return new PageQueryResult<TItem>(
-                countResponse.Resource,
+                countResponse?.Resource ?? null,
                 pageNumber,
                 pageSize,
                 result.Items.AsReadOnly(),
-                result.Charge + countResponse.RequestCharge);
+                result.Charge + countResponse?.RequestCharge ?? 0);
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/DefaultRepository.cs:445]: Add charge again I think.

**Review by mumby0168 — APPROVED** [review]
Looks good to me. I much prefer this to be an opt-in feature given the potential cost implications on large data sets. @IEvangelist are you cool with this?

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**